### DISCUSSION TITLE
Any good method to determine current player is standing on sth.?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1068] [Q&A] [answered]
Hi:

Is there any built-in mechanism to determine whether player is standing on sth.?
Currently we keep the y-velocity of previous frame and decide whether character is standing on sth. by checking whether y-velocity of current frame and previous frame are ~0, if the answer is yes then the character is allowed to jump. 
If no, then he won't jump
But I am not sure whether is the best practice or correct way to do it
so is there any built-in mechanism for us to check if the player is standing on sth.? 

Thanks

**Comment by AlmasB** [comment, marked as answer]
Hi,

If the entity in question uses PhysicsComponent, then you can add a ground sensor:

https://github.com/AlmasB/FXGL/blob/8bda93136a3ed9b0d297ead25ae161c84ff0ae57/fxgl-samples/src/main/java/sandbox/PlatformerSample.java#L201-L205

Then check:

https://github.com/AlmasB/FXGL/blob/8bda93136a3ed9b0d297ead25ae161c84ff0ae57/fxgl-samples/src/main/java/sandbox/PlatformerSample.java#L122-L124

If the entity does not use PhysicsComponent, then what you are doing is probably a good approach.
